src/EditorFeatures/Core/RenameTracking/RenameTrackingTaggerProvider.TrackingSession.cs

```diff
@@ -108,22 +108,24 @@ private void QueueUpdateToStateMachine(StateMachine stateMachine, Task task)
         {
             var asyncToken = _asyncListener.BeginAsyncOperation($"{GetType().Name}.{nameof(QueueUpdateToStateMachine)}");
 
-            task.SafeContinueWithFromAsync(async t =>
-               {
-                   await _threadingContext.JoinableTaskFactory.SwitchToMainThreadAsync(alwaysYield: true, _cancellationToken).NoThrowAwaitable();
-
-                   // Avoid throwing an exception in this common case
-                   if (_cancellationToken.IsCancellationRequested)
-                       return;
-
-                   if (_isRenamableIdentifierTask.Result != TriggerIdentifierKind.NotRenamable)
-                   {
-                       stateMachine.OnTrackingSessionUpdated(this);
-                   }
-               },
-               _cancellationToken,
-               TaskContinuationOptions.OnlyOnRanToCompletion | TaskContinuationOptions.ExecuteSynchronously,
```

Comment:
the prior logic was 'OnlyOnRanToCompletion' which is what 'await' does anyways. So the conversion is sound.